### PR TITLE
Store slab-size change info in DataBox instead of inboxes

### PR DESCRIPTION
**Pull request by wthrowe**
It used to be that Events could not modify the DataBox, so the only way to store information was to synchronously send yourself a message to an inbox.  This is no longer true, so this changes to a cleaner organization.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/ControlSystem/Actions/LimitTimeStep.hpp [hunk 1]
```diff
@@ -14,7 +14,7 @@
 #include "Parallel/ArrayComponentId.hpp"
 #include "Parallel/Callback.hpp"
 #include "Parallel/GlobalCache.hpp"
-#include "Time/ChangeSlabSize.hpp"
+#include "Time/ChangeSlabSize/ChangeSlabSize.hpp"
```
Comment (kidder) [inline src/ControlSystem/Actions/LimitTimeStep.hpp:17]: commit message has a typo (mush -> much)

File: src/Time/ChangeSlabSize/Projector.hpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            children_items.begin()->second);
+    for (const auto& child : children_items) {
+      ASSERT(*number_of_expected_messages ==
+                 tuples::get<::Tags::ChangeSlabSize::NumberOfExpectedMessages>(
+                     child.second),
+             "Different elements expecting different numbers of "
+             "ChangeSlabSize messages.");
+    }
+  }
+};
```
Comment (kidder) [inline src/Time/ChangeSlabSize/Projector.hpp]: I believe this projector is currently the same as the generic projector `ParallelAlgorithms/Amr/Projectors/CopyFromCreatorOrLeaveAsIs.hpp` 
I leave it up to you as to whether you want to keep this or use the generic projector...

**Review by kidder — CHANGES_REQUESTED** [review]
- one typo in a commit message, which you can squash
- one optional change to consider

**Comment by wthrowe** [comment]
Used the generic projector and added a commit adding documentation to the AMR namespaces so they show up.

Out of scope, but the metavariables seem like the wrong place for these things.  The projectors should be specified in the same place that the tags are added to the DataBox.

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
> Out of scope, but the metavariables seem like the wrong place for these things. The projectors should be specified in the same place that the tags are added to the DataBox.

Yes, we should probably have a core-dev discussion about this...

**Pull request merged**